Source/DataSources/CallbackProperty.js

```diff
@@ -104,8 +104,8 @@ CallbackProperty.prototype.equals = function (other) {
  * A function that returns the value of the property.
  * @callback CallbackProperty.Callback
  *
- * @param {JulianDate} [time] The time for which to retrieve the value.
- * @param {Object} [result] The object to store the value into, if omitted, a new instance is created and returned.
- * @returns {Object} The modified result parameter or a new instance if the result parameter was not supplied or is unsupported.
+ * @param {JulianDate} time The time for which to retrieve the value.
```

Comment:
I think the `time` parameter _is_ optional.

Reply:
This is a kind of a mess.

Since this is a callback-function signature, what we're actually saying here is whether or not the function is guaranteed to be invoked with a time value, or whether the function must handle the case where `time` is undefined.  The callback is invoked by passing through the `time` argument when `getValue` is called.

[Here](https://github.com/thw0rted/cesium/blob/more-typescript-redo/Source/DataSources/CallbackProperty.js#L52) is the `time` argument for `CallbackProperty#getValue` -- it looks like this was erroneously copy-pasted from `ConstantProperty`, since obviously a ConstantProperty does not vary with time but a CallbackProperty (probably?) could.  *If* the `time` argument for CallbackProperty should in fact be required, then the callback signature can be marked as non-optional, and callbacks can safely assume `time` is not undefined.  My inclination is to resolve this by marking the `time` argument to `getValue` as required as well.  That would be in line with the [argument in the base Property interface](https://github.com/thw0rted/cesium/blob/more-typescript-redo/Source/DataSources/Property.js#L55), which is required.

Reply:
OK, I think I understand your point, thanks! I agree the parameter for the `getValue` function should be required.